pkg/analyzer/analyzer.go

```diff
@@ -0,0 +1,1009 @@
+			//package node
+			//sort namespaces
+			sort.Sort(packageNameSpaces(dependency.Package.Namespaces))
+			message := dependency.Package.Type
+			for _, namespace := range dependency.Package.Namespaces {
+				message += namespace.Namespace
+				sort.Sort(packageNameSpacesNames(namespace.Names))
+				for _, name := range namespace.Names {
+					message += name.Name
+					sort.Sort(packageNameSpacesNamesVersions(name.Versions))
+					for _, version := range name.Versions {
+						message += version.Version
+						message += version.Subpath
+						sort.Sort(packageNameSpacesNamesVersionsQualifiers(version.Qualifiers))
+						for _, outlier := range version.Qualifiers {
+							message += outlier.Key
+							message += outlier.Value
+						}
+					}
+				}
+			}
+
+			if message == "" {
+				return g, fmt.Errorf("encountered empty message for hashing")
+			}
+
+			hashValPackage := nodeHasher([]byte(message))
+			_, err := g.Vertex(hashValPackage)
+
+			if err != nil { //node does not exist
+				AddGraphNode(g, hashValPackage, "black") // so, create a node
+				AddGraphEdge(g, "HasSBOM", hashValPackage, "black")
+				//set attributes here
+				if !(SetNodeAttribute(g, hashValPackage, "nodeType", "Package") &&
+					SetNodeAttribute(g, hashValPackage, "data", dependency.Package)) {
+					return g, fmt.Errorf("error setting package node attribute(s)")
+				}
+			}
+
+			//dependencyPackage node
+			sort.Sort(packageNameSpaces(dependency.DependencyPackage.Namespaces))
+			message = dependency.DependencyPackage.Type
+			for _, namespace := range dependency.DependencyPackage.Namespaces {
+				message += namespace.Namespace
+				sort.Sort(packageNameSpacesNames(namespace.Names))
+				for _, name := range namespace.Names {
+					message += name.Name
+					sort.Sort(packageNameSpacesNamesVersions(name.Versions))
+					for _, version := range name.Versions {
+						message += version.Version
+						message += version.Subpath
+						sort.Sort(packageNameSpacesNamesVersionsQualifiers(version.Qualifiers))
+						for _, outlier := range version.Qualifiers {
+							message += outlier.Key
+							message += outlier.Value
+						}
+					}
+				}
+			}
+
+			hashValDependencyPackage := nodeHasher([]byte(message))
+			_, err = g.Vertex(hashValDependencyPackage)
+
+			if err != nil { //node does not exist
+				AddGraphNode(g, hashValDependencyPackage, "black")
+				if !(SetNodeAttribute(g, hashValDependencyPackage, "nodeType", "DependencyPackage") &&
+					SetNodeAttribute(g, hashValDependencyPackage, "data", dependency.DependencyPackage)) {
+					return g, fmt.Errorf("error setting dependency package node attribute(s)")
+				}
+			}
+
+			AddGraphEdge(g, hashValPackage, hashValDependencyPackage, "black")
+		}
+	}
+	return g, nil
+}
+func nodeHasher(value []byte) string {
+	hash := sha256.Sum256(value)
+	return hex.EncodeToString(hash[:])
+}
+
+func AddGraphNode(g graph.Graph[string, *Node], _ID, color string) {
```

Comment:
what is color used for? Is it the identity of an edge of just for analysis purpose? can it be passed it as an option on the node for extensibility? 

`_ID` should be id to follow golang style?